node/worker/worker.go

```diff
@@ -67,6 +67,9 @@ func (w *Worker) SelectTransactionsForNewBlock(txs types.Transactions, maxNumTxs
 			}
 		}
 	}
+	if err := w.UpdateCurrent(coinbase); err != nil {
```

Comment:
In the past, before disabling the tx for mainnet - the logic here is to call getTransactionsForNewBlock first https://github.com/harmony-one/harmony/blob/before_sc/node/node_newblock.go#L140, then call Worker.CommitTransactions

And inside getTransactionsForNewBlock, we call w.UpdateCurrent()

During refactoring for mainnet - we removed w.UpdateCurrent() out of getTransactionsForNewBlock and moved it after CommitTransactions.

I did a lot of debug and found out we need to move it back.

I also found CommitTransactions redundant to our code - and tried to removed it but removing that made the tx logic not working again.

Since I had to progress on SDK, integration test, SDK workflow i did continue the investigation to answer why removing CommitTransactions made tx logic not working.